lib/kitchen/driver/aws/instance_generator.rb

```diff
@@ -62,22 +62,38 @@ def ec2_instance_data # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
           end
 
           if config[:security_group_ids].nil? && config[:security_group_filter]
-            security_group = ::Aws::EC2::Client
-                .new(region: config[:region]).describe_security_groups(
-                filters: [
-                    {
-                        name: "tag:#{config[:security_group_filter][:tag]}",
-                        values: [config[:security_group_filter][:value]],
-                    },
+            security_groups = []
+            filters = [config[:security_group_filter]].flatten
+            filters.each do |sg_filter|
+              r = {}
+              if sg_filter[:name]
+                r[:filters] = [
+                  {
+                    name: "group-name",
+                    values: [sg_filter[:name]],
+                  },
                 ]
-            )[0][0]
+              end
 
-            if security_group
-              config[:security_group_ids] = [security_group.group_id]
-            else
-              raise "The group tagged '#{config[:security_group_filter][:tag]} " +
-                "#{config[:security_group_filter][:value]}' does not exist!"
+              if sg_filter[:tag]
+                r[:filters] = [
+                  {
+                    name: "tag:#{sg_filter[:tag]}",
+                    values: [sg_filter[:value]],
+                  },
+                ]
+              end
+
+              security_group = ::Aws::EC2::Client.new(region: config[:region]).describe_security_groups(r)[0][0]
+
+              if security_group
+                security_groups.push(security_group.group_id)
+              else
+                raise "The group tagged '#{sg_filter[:tag]} " +
+                  "#{sg_filter[:value]}' does not exist!"
```

Comment:
I think here we are assuming that `tag` and `value` are always provided but now we accept `name` so I think we need to update this error to know when we are looking for a group-name or a tag.

Reply:
Agreed.  We could probably just spit out the entire object instead.  Will update the PR tmm.

Reply:
Code has been updated, error output now looks like
```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [A Security Group matching the following filter could not be found!{:tag=>"Name", :value=>"MyCoolGroup"}] on default-centos7
```